### PR TITLE
Remove Touch Interaction

### PR DESCRIPTION
**Pull request by philippfromme**
Required by https://github.com/bpmn-io/bpmn-js/pull/2079
Closes https://github.com/bpmn-io/diagram-js/issues/796

### CODE REVIEW

**Review by nikku — COMMENTED** [review]
My question on this PR is:

* What options do contributors have to re-add touch support, given that we remove _everything_ from the core? If there is some, then I'll be fine. If there is none, that essentially means we close the door to touch support completely, which blocks users from building a proper working touch support.

File: lib/features/connect/Connect.js [hunk 1]
```diff
@@ -122,7 +122,7 @@ export default function Connect(eventBus, dragging, modeling, rules) {
   /**
    * Start connect operation.
    *
-   * @param {MouseEvent|TouchEvent} event
```
[the remark this answers is not in the document]
Comment (philippfromme) [reply lib/features/connect/Connect.js]: My idea was that all of these things could be added back in through an extension. But to validate that I'd have to build that extension. So for now I've reverted the changes.

File: lib/features/dragging/Dragging.js [hunk 2]
```diff
@@ -174,8 +167,7 @@ export default function Dragging(eventBus, canvas, selection, elementRegistry) {
       assign(
         {},
         dragContext.payload,
-        dragContext.data,
-        { isTouch: dragContext.isTouch }
```
[the remark this answers is not in the document]
Comment (philippfromme) [reply lib/features/dragging/Dragging.js]: See https://github.com/bpmn-io/diagram-js/pull/845#discussion_r1461580308

**Comment by philippfromme** [comment]
> My question on this PR is:
> 
> * What options do contributors have to re-add touch support, given that we remove _everything_ from the core? If there is some, then I'll be fine. If there is none, that essentially means we close the door to touch support completely, which blocks users from building a proper working touch support.

My idea was that all of these things could be added back in through an extension. But to validate that I'd have to build that extension. So for now I've reverted the changes.

**Comment by nikku** [comment]
Let's merge these aspects. Gives users a chance to plug-in touch interaction (again).

**Review by nikku — APPROVED** [review]

**Comment by tony2y** [comment]
I will use it on mobile devices in my work, so how can I expand it if I want to add touch support now? Do you have your own examples of how to expand?
@philippfromme @nikku

**Comment by tony2y** [comment]
image
Why can the demonstration address be moved after switching to mobile mode, and what do I need to do to achieve the same effect?

**Comment by barmac** [comment]
BTW I checked how demo.bpmn.io works on iPhone 13 Mini and it seems to be usable even after the changes. The only parts which don't work right now are the color picker and the append-anything menu.

**Comment by vesper8** [comment]
@barmac Are you sure it works? Because I tried the demo on both ipad and iphone and it doesn't work at all. You cannot pan, you cannot drag.

Is there a work-in-progress re-implementation of touch support, as an extension, anywhere in the wild yet?

**Comment by nikku** [comment]
> Is there a work-in-progress re-implementation of touch support, as an extension, anywhere in the wild yet?

There is no work-in-progress re-implementation yet, but if you'd like to contribute one we're happy to help.

**Comment by barmac** [comment]
I tested now and only some interactions work on my iphone. I can place new elements, but cannot move the existing ones, and I cannot move around the diagram. I'd classify the tool as unusable on a touch-only device.

**Pull request merged**